v2/yarpchttp/yarpchttpfx/yarpchttpfx.go

```diff
@@ -0,0 +1,136 @@
+package yarpchttpfx
+
+import (
+	"context"
+	"net/url"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"go.uber.org/config"
+	"go.uber.org/fx"
+	yarpc "go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpchttp"
+	"go.uber.org/zap"
+)
+
+const (
+	_name                     = "yarpchttpfx"
+	_inboundConfigurationKey  = "yarpchttp.inbounds"
+	_outboundConfigurationKey = "yarpchttp.outbounds"
+)
+
+// Module produces yarpchttp clients and starts yarpchttp inbounds.
+var Module = fx.Options(
+	fx.Provide(NewClients),
+	fx.Invoke(StartInbounds),
+)
+
+// InboundConfig is the configuration for starting yarpchttp inbounds.
+type InboundConfig struct {
+	Address string `yaml:"address"`
+}
+
+// StartInboundsParams defines the dependencies of this module.
+type StartInboundsParams struct {
+	fx.In
+
+	Lifecycle fx.Lifecycle
+	Router    yarpc.Router
+	Provider  config.Provider
```

Comment:
Discussed offline -
Yeah this sounds good! One concern we have is that this would result in a lot of configuration structs floating around in the Fx application container. We can revisit this once we roll out more `<transport>fx` packages and determine whether or not the configuration should be loaded implicitly.